src/components/modal/modal.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import {SBComponent} from 'stylable-react-component';
+import {root} from 'wix-react-tools';
+import {Portal} from '../../../src';
+import {noop} from '../../utils';
+import {enableScrolling, stopScrolling} from '../../utils/stop-scrolling';
+import styles from './modal.st.css';
+
+export interface RequestCloseEvent {
+    source: string;
+}
+
+export interface ModalProps extends React.HTMLAttributes<HTMLDivElement> {
+    isOpen: boolean;
+    onRequestClose?({source}: RequestCloseEvent): void;
```

Comment:
This is a bit confusing, what about something like:
onRequestClose?: (event: RequestCloseEvent) => void;